### PR TITLE
astparser autotools build

### PR DESCRIPTION
**Pull request by omusil24**

### CODE REVIEW

**Comment by omusil24** [comment]
@davidnich when adding the autotools support, I got some errors at first during the build:
```
QC_AstParser.qpp:88:16: error: conversion from ‘std::nullptr_t’ to ‘QoreValue’ is ambiguous
         return nullptr;
                ^
In file included from /usr/local/include/qore/Qore.h:55:0,
                 from AstParserHolder.h:34,
                 from AstParserHolder.cpp:28,
                 from astparser_scu.cpp:7:
/usr/local/include/qore/QoreValue.h:164:14: note: candidate: QoreValue::QoreValue(const AbstractQoreNode*)
    DLLEXPORT QoreValue(const AbstractQoreNode* n);
              ^
/usr/local/include/qore/QoreValue.h:155:14: note: candidate: QoreValue::QoreValue(AbstractQoreNode*)
    DLLEXPORT QoreValue(AbstractQoreNode* n);
```

I solved it in a4bb189223016cdd98a6677fdbf3f53ae9a328a9 by adding casts to `AbstractQoreNode*` to all the nullptrs and it's really ugly now. Is there any other way? Or a way how to fix it at the QoreValue level? (probably removing one of the constructor overloads)

**Comment by davidnich** [comment]
@omusil24 please do not return a `nullptr` value for a `QoreValue`; instead just use: `return QoreValue();`

**Comment by omusil24** [comment]
Ah, ok. I was just thinking about it.

File: modules/astparser/src/QC_AstParser.qpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ AstParser::constructor() {
 *AstTree AstParser::parseFile(string filename) {
     ASTTree* tree = aph->parseFile(filename->getBuffer());
     if (!tree)
-        return nullptr;
+        return (AbstractQoreNode*) nullptr;
```
Comment (davidnich) [inline modules/astparser/src/QC_AstParser.qpp]: please change this to `QoreValue()` - also in the other locations, thx - a global search and replace should do it it one go

**Review by davidnich — CHANGES_REQUESTED** [review]

**Comment by omusil24** [comment]
@davidnich do you have any idea what's happening here: https://hq.qoretechnologies.com/jenkins/job/qore-cmake/455/console
It seems that in the cmake build, QPP doesn't use QoreValue but the old node pointers instead.

**Comment by davidnich** [comment]
@omusil24 are you calling qpp as `qpp -V` to tell it to use the QoreValue API in the generated code?

**Comment by omusil24** [comment]
@davidnich all done...

**Review by davidnich — APPROVED** [review]

**Pull request merged**